test/ash_authentication/strategies/api_key_test.exs

```diff
@@ -71,6 +71,21 @@ defmodule AshAuthentication.Strategy.ApiKeyTest do
       assert conn.assigns.current_user.id == user.id
     end
 
+    test "succeeeds when API key is present in header with regex prefix", %{
+      plaintext_api_key: plaintext_api_key,
+      user: user
+    } do
+      opts = ApiKeyPlug.init(resource: Example.User, source: :header, header_prefix: ~r/bearer/i)
```

Comment:
This is technically a bug I believe, as you'd want ^bearer. Perhaps we should validate that the regex starts with `^`

Reply:
Oops, good catch! 😬

I've updated with a validation and a test. Let me know if you'd like any other changes!